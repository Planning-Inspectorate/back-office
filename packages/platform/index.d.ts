import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
