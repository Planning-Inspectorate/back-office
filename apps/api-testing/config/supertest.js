// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
