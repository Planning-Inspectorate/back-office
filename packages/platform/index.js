export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
