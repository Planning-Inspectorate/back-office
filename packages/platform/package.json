{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
