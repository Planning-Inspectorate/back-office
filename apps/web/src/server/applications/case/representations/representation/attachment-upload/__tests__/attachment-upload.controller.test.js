import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../../../testing/index.js';
import { representationFixture } from '../../../__fixtures__/representations.fixture.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const mockCaseReference = { title: 'mock title', status: 'in test', reference: 'mock reference' };
const nocks = () => {
	nock('http://test/').get('/applications/1').reply(200, mockCaseReference);
	nock('http://test/').get('/applications/1/representations/1').reply(200, representationFixture);
	nock('http://test/')
		.get('/applications/1/folders')
		.reply(200, [
			{
				id: 10,
				displayNameEn: 'Relevant representations'
			}
		]);
};

describe('Representation attachment-upload page', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	const baseUrl =
		'/applications-service/case/1/relevant-representations/attachment-upload?repId=1&repType=represented';

	describe('GET /applications-service/case/1/relevant-representations/attachment-upload', () => {
		beforeEach(async () => {
			nocks();
			await request.get('/applications-service/');
		});

		it('should render the page and have the lookup fields', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain(
				'data-next-page-url="/applications-service/case/1/relevant-representations/1/representation-details"'
			);
			expect(element.innerHTML).toContain('data-folder-id="10"');
			expect(element.innerHTML).toContain('data-case-id="1"');
		});
	});
});
