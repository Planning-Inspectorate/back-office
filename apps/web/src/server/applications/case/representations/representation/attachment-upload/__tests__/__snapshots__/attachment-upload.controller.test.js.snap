// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Representation attachment-upload page GET /applications-service/case/1/relevant-representations/attachment-upload should render the page and have the lookup fields 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\"><span class=\\"govuk-caption-l\\">Attachment upload</span>
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-full\\">
            <div class=\\"govuk-grid-row pins-file-upload\\" data-next-page-url=\\"/applications-service/case/1/relevant-representations/1/representation-details\\"
            data-case-id=\\"1\\" data-multiple=\\"true\\" data-folder-id=\\"10\\" data-allowed-types=\\".pdf,application/pdf,.doc,application/msword,.docx,application/vnd.openxmlformats-officedocument.wordprocessingml.document,.ppt,application/vnd.ms-powerpoint,.pptx,application/vnd.openxmlformats-officedocument.presentationml.presentation,.xls,application/vnd.ms-excel,.xlsx,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,.jpg,image/jpeg,.jpeg,image/jpeg,.mpeg,video/mpeg,.mp3,audio/mpeg,.mp4,video/mp4,.mov,video/quicktime,.png,image/png,.tif,image/tiff,.tiff,image/tiff,\\">
                <div class=\\"top-errors-hook\\"></div>
                <h1 class=\\"govuk-heading-m govuk-grid-column-full\\">Planning Inspectorate Applications</h1>
                <form class=\\"govuk-grid-column-two-thirds\\">
                    <div class=\\"pins-file-upload--container\\">
                        <div class=\\"govuk-body colour--secondary pins-file-upload--instructions\\">
                            <label class='pins-file-upload--text' for=\\"upload-file-1\\">Choose a single or multiple files to upload.</label><span class='pins-file-upload--text'> Your file must be PDF, DOC, DOCX, PPT, PPTX, XLS, XLSX, JPG, JPEG, MPEG, MP3, MP4, MOV, PNG, TIF, 	or TIFF
.</span><span class='pins-file-upload--text'>The total of your uploaded files must be smaller than 1 GB.</span>
                            <div                             class=\\"middle-errors-hook\\"></div>
                    </div>
                    <div class=\\"display--flex\\">
                        <input class=\\"display--none\\" id=\\"upload-file-1\\" accept=\\".pdf,application/pdf,.doc,application/msword,.docx,application/vnd.openxmlformats-officedocument.wordprocessingml.document,.ppt,application/vnd.ms-powerpoint,.pptx,application/vnd.openxmlformats-officedocument.presentationml.presentation,.xls,application/vnd.ms-excel,.xlsx,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,.jpg,image/jpeg,.jpeg,image/jpeg,.mpeg,video/mpeg,.mp3,audio/mpeg,.mp4,video/mp4,.mov,video/quicktime,.png,image/png,.tif,image/tiff,.tiff,image/tiff,\\"
                        type=\\"file\\" name=\\"files\\" value=\\"Choose file\\" aria-controls=\\"file-list-1\\"
                        multiple>
                        <button type='button' class=\\"pins-file-upload--button govuk-button\\">Choose file</button><span role=\\"status\\" aria-live=\\"assertive\\" class='govuk-body pins-file-upload--counter'> No file chosen</span>
                    </div>
                    <div id=\\"file-list-1\\">
                        <h2 class=\\"display--sr-only\\" id=\\"file-list-title-1\\">List of files to upload</h2>
                        <ul class=\\"pins-file-upload--files-rows\\" aria-describedby=\\"file-list-title-1\\"
                        aria-live=\\"polite\\"></ul>
                    </div>
            </div>
            <div class=\\"govuk-inset-text\\">Saving files will default them to ‘Unredacted’ and ‘Not checked’</div>
            <div             class=\\"display--flex\\">
                <button class=\\"govuk-button pins-file-upload--submit\\" data-module=\\"govuk-button\\"
                id=\\"submit-button-1\\">Save and continue</button>
                <div class=\\"progress-hook\\"></div>
        </div>
        </form>
    </div><a href=\\"/applications-service/case/1/relevant-representations/1/representation-details\\"> Skip this step</a>
    </div>
    </div>
</main>"
`;
